// @mui/x-tree-view/legacy/TreeView/TreeViewContext.js
import * as React from 'react';
/**
 * @ignore - internal component.
 */
export var TreeViewContext = /*#__PURE__*/React.createContext({
  registerNode: function registerNode() {},
  unregisterNode: function unregisterNode() {},
  isFocused: function isFocused() {
    return false;
  },
  isSelected: function isSelected() {
    return false;
  },
  isExpanded: function isExpanded() {
    return false;
  },
  isExpandable: function isExpandable() {
    return false;
  },
  isDisabled: function isDisabled() {
    return false;
  },
  mapFirstChar: function mapFirstChar() {},
  unMapFirstChar: function unMapFirstChar() {},
  focus: function focus() {},
  toggleExpansion: function toggleExpansion() {},
  selectNode: function selectNode() {},
  selectRange: function selectRange() {},
  multiSelect: false,
  disabledItemsFocusable: false,
  treeId: undefined,
  icons: {
    defaultCollapseIcon: null,
    defaultExpandIcon: null,
    defaultParentIcon: null,
    defaultEndIcon: null
  }
});
if (process.env.NODE_ENV !== 'production') {
  TreeViewContext.displayName = 'TreeViewContext';
}
